From 5350adfd47c657a9f6a2e60ca3d67bc2fe4d3f3a Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Mon, 22 Sep 2025 11:09:44 +0200
Subject: [PATCH] Fix IDL of SVG spec

HTMLHyperlinkElementUtils: https://github.com/w3c/svgwg/issues/312

Also drop `SVGPathElement`, now that SVG Paths is also being crawled, pending
clarification of status between SVG 2 and SVG Paths.

Constructor patch is a temporary fix until svgwg.org correctly reflects the
latest version of the Editor's Draft.
---
 ed/idl/SVG.idl | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/ed/idl/SVG.idl b/ed/idl/SVG.idl
index d046678667..a344b5df96 100644
--- a/ed/idl/SVG.idl
+++ b/ed/idl/SVG.idl
@@ -314,8 +314,9 @@ interface mixin SVGElementInstance {
   [SameObject] readonly attribute SVGUseElement? correspondingUseElement;
 };
 
-[Constructor(Animation source, Animatable newTarget), Exposed=Window]
+[Exposed=Window]
 interface ShadowAnimation : Animation {
+  constructor(Animation source, (Element or CSSPseudoElement) newTarget);
   [SameObject] readonly attribute Animation sourceAnimation;
 };
 
@@ -418,10 +419,6 @@ interface SVGAnimatedPreserveAspectRatio {
   [SameObject] readonly attribute SVGPreserveAspectRatio animVal;
 };
 
-[Exposed=Window]
-interface SVGPathElement : SVGGeometryElement {
-};
-
 [Exposed=Window]
 interface SVGRectElement : SVGGeometryElement {
   [SameObject] readonly attribute SVGAnimatedLength x;
@@ -671,7 +668,20 @@ interface SVGAElement : SVGGraphicsElement {
 };
 
 SVGAElement includes SVGURIReference;
-SVGAElement includes HTMLHyperlinkElementUtils;
+
+// Inline HTMLHyperlinkElementUtils except href, which conflicts.
+partial interface SVGAElement {
+  readonly attribute USVString origin;
+  [CEReactions] attribute USVString protocol;
+  [CEReactions] attribute USVString username;
+  [CEReactions] attribute USVString password;
+  [CEReactions] attribute USVString host;
+  [CEReactions] attribute USVString hostname;
+  [CEReactions] attribute USVString port;
+  [CEReactions] attribute USVString pathname;
+  [CEReactions] attribute USVString search;
+  [CEReactions] attribute USVString hash;
+};
 
 [Exposed=Window]
 interface SVGViewElement : SVGElement {};
-- 
2.51.0

